src/realm/array_direct.hpp

```diff
@@ -199,13 +199,14 @@ class UnalignedWordIter {
         auto first_word = m_word_ptr[0];
         uint64_t result = first_word >> m_in_word_offset;
         // note: above shifts in zeroes
-        if (m_in_word_offset + num_bits <= 64)
-            return result;
-        // if we're here, in_word_offset > 0
-        auto first_word_size = 64 - m_in_word_offset;
-        auto second_word = m_word_ptr[1];
-        result |= second_word << first_word_size;
-        // note: above shifts in zeroes below the bits we want
+        if (m_in_word_offset + num_bits > 64) {
+            // if we're here, in_word_offset > 0
+            auto first_word_size = 64 - m_in_word_offset;
+            auto second_word = m_word_ptr[1];
+            result |= second_word << first_word_size;
+            // note: above shifts in zeroes below the bits we want
+        }
+        _bump(num_bits);
         return result;
     }
     uint64_t get_with_unsafe_prefetch(size_t num_bits)
```

Comment:
You are now fetching and moving the iterator in this method, probably we should rename it? Even though, I don't expect anybody to use the iterator. Although, it can be useful if we want to access the data directly.